Source/WebKit/UIProcess/Cocoa/VideoPresentationManagerProxy.mm

```diff
@@ -1060,6 +1084,8 @@ - (BOOL)prefersStatusBarHidden
         m_mockPictureInPictureWindowSize.setHeight(DefaultMockPictureInPictureWindowWidth / videoDimensions.aspectRatio());
         return;
     }
+    if (videosInElementFullscrenChanged)
+        this->videosInElementFullscreenChanged();
```

Comment:
Maybe we need a different name here. The videos in element fullscreen haven't changed, but we do need to re-evaluate the list of videos. Maybe `updateVideosInElementFullscreen()`?

Reply:
It is really to say that one of the videos in the element fullscreen changed. So we need to recheck which one is the most visible. 

I guess I could store the context Id of the last video we made primary and only recheck if that video changed